spec/integration/stats_partition_eraser_job_spec.rb

```diff
@@ -0,0 +1,59 @@
+require_relative '../spec_helper'
+
+RSpec.describe ThreeScale::Backend::Stats::PartitionEraserJob do
+  let(:service_id) { '123456' }
+  let(:applications) { %w[1] }
+  let(:metrics) { %w[10] }
+  let(:users) { %w[] }
+  let(:from) { Time.new(2002, 11, 31).to_i }
+  let(:to) { from }
+  let(:offset) { 5 }
+  let(:length) { 10 }
+  let(:job_params) do
+    {
+      service_id: service_id,
+      applications: applications,
+      metrics: metrics,
+      users: users,
+      from: from,
+      to: to
+    }
+  end
+  let(:stats_queue) { :stats }
+  let(:storage) { ThreeScale::Backend::Storage.instance }
+  let(:keys) { ThreeScale::Backend::Stats::KeyGenerator.new(job_params).keys }
+
+  before :each do
+    storage.flushdb
+    ThreeScale::Backend::Worker::QUEUES.each { |queue| Resque.remove_queue(queue) }
+    # populate all keys
+    keys.each { |key| storage.set(key, 1) }
+  end
+
+  it 'partition keys are deleted' do
+    without_resque_spec do
+      Resque.enqueue(described_class, Time.now.getutc.to_f, service_id, applications,
+                     metrics, users, from, to, offset, length, nil)
+      expect(Resque.size(stats_queue)).to eq 1
+      # Try to process the job.
+      ThreeScale::Backend::Worker.work(one_off: true)
+      expect(keys.drop(offset).take(length).count).to be > 0
+      keys.drop(offset).take(length).each do |key|
+        expect(storage.get(key)).to be_nil
```

Comment:
Nitpick: you can use `exists` instead of `get` for this. Also, this can be rewritten to use a single expect.

```ruby
keys_to_be_deleted = keys.drop(offset).take(length)
expect(keys_to_be_deleted.none? { |key| storage.exists(key) })
```